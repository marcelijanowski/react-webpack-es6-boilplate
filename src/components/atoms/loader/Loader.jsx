import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Loader extends PureComponent {
  render() {
    const { text } = this.props;
    return (<div className="hackernewslist-state__loading">{text}</div>);
  }
}

Loader.propTypes = {
  text: PropTypes.string.isRequired
};


export default Loader;