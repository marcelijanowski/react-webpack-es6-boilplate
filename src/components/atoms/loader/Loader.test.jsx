import React, { PureComponent } from 'react'
import renderer from 'react-test-renderer';

import Loader from './Loader';
import { mockRef, mountComponent } from '../../../../test/testHelper';

describe('HackernewsList container', () => {
  it('renders a snapshot', () => {
    const tree = renderer.create(<Loader text="Data is loading..." />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})