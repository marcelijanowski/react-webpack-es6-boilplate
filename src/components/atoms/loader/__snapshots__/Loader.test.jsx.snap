// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HackernewsList container renders a snapshot 1`] = `
<div
  className="hackernewslist-state__loading"
>
  Data is loading...
</div>
`;
